main.go

```diff
@@ -1,48 +1,55 @@
 package main
 
 import (
-	"encoding/json"
+	"context"
 	"flag"
 	"log"
-	"net/http"
-	"strings"
+
+	"github.com/Clever/resolve-ip/gen-go/models"
+	"github.com/Clever/resolve-ip/gen-go/server"
+	"gopkg.in/Clever/kayvee-go.v5/logger"
 )
 
+var addr = flag.String("addr", ":8080", "Address to listen at")
 var path = flag.String("path", "./GeoLiteCity", "path to Geo Lite City db")
-var port = flag.String("port", "80", "port to listen on")
+
+type handler struct {
+	db *GeoDB
+}
+
+func (h handler) HealthCheck(ctx context.Context) error {
+	return nil
+}
+
+func (h handler) LocationForIP(ctx context.Context, i *models.LocationForIPInput) (*models.IP, error) {
+	latlon, err := h.db.Lookup(i.Ip)
+	if err == ErrIPMissing {
+		return nil, models.LocationForIP404Output{}
+	}
+	if err == ErrBadIP {
+		return nil, models.DefaultBadRequest{Msg: err.Error()}
+	}
+	if err != nil {
+		return nil, models.DefaultInternalError{Msg: err.Error()}
+	}
```

Comment:
When @kvigen and I drew up the `Default*` types, they weren't really meant to be used within handlers. They were meant for internal use by wag itself: `DefaultBadRequest` is returned when wag detects invalid requests, and `DefaultInternalError` is returned when the handler returns an `error` not defined in the spec.

(@kvigen maybe we should make these private to the framework?)

I think a better approach here would be to (1) define the `ErrBadIP` 400 error in the swagger spec, and then return that type instead of `models.DefaultBadRequest`, and (2) just `return nil, err` for other, unknown errors.